beeline-core/src/main/java/io/honeycomb/beeline/tracing/propagation/CompositeHttpHeaderPropagator.java

```diff
@@ -54,10 +74,10 @@ public PropagationContext decode(Map<String, String> encodedTrace) {
      * @return a valid AWS http header value - empty if required IDs are missing or input is null.
      */
     @Override
-    public Optional<Map<String, String>> encode(PropagationContext context) {
-        Map<String, String> headers = new HashMap<>();
-        for (PropagationCodec<Map<String, String>> codec : codecs) {
-            Optional<Map<String, String>> codecheaders = codec.encode(context);
+    public Optional<Map<String, String>> encode(final PropagationContext context) {
+        final Map<String, String> headers = new HashMap<>();
+        for (final PropagationCodec<Map<String, String>> codec : codecs) {
+            final Optional<Map<String, String>> codecheaders = codec.encode(context);
```

Comment:
Being a fairweather Java programmer, I'm unfamiliar with the reasons for adding final to params, etc. Is it just to ensure immutability? e.g. Scala uses `val` for this purpose, but I don't think there's an equivalent in Java.

Reply:
Yeah, `final` just marks a parameter as immutable and the function receives a copy instead of the real value so it's effectively immutable from the outside perspective.

Java 10+ does have the `var` keyword, but I think that just tries to invert type inference so you don't need to declare the full property type on the left-hand side. However, we can't use it because we support < Java 10 😢 .